### PR TITLE
v1.13 Backports 2023-09-21

### PR DESCRIPTION
**Pull request by doniacld**
* [x] #23093 (@obaranov1)  (verified by @giorio94)
 * [x] #27996 (@jschwinger233)
 * [x] #27870 (@joamaki)
 * [x] #28131 (@carnerito)
 * [x] #28122 (@qmonnet)
 * [x] #28133 (@julianwiedmann)
 * [x] #28161 (@qmonnet)
 * [x] #28145 (@margamanterola)
 * [x] #28172 (@qmonnet)
 * [x] #28200 (@sayboras)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
for pr in 23093 27996 27870 28131 28122 28133 28161 28145 28172 28200; do contrib/backporting/set-labels.py $pr done 1.13; done
```
or with
```
make add-labels BRANCH=v1.13 ISSUES=23093,27996,27870,28131,28122,28133,28161,28145,28172,28200
```

### CODE REVIEW

File: .github/workflows/conformance-gateway-api.yaml [hunk 1]
```diff
@@ -16,15 +16,15 @@ on:
       extra-args:
         description: "[JSON object] Arbitrary arguments passed from the trigger comment via regex capture group. Parse with 'fromJson(inputs.extra-args).argName' in workflow."
         required: false
-        default: '{}'
```
Comment (sayboras) [inline .github/workflows/conformance-gateway-api.yaml]: this seems to have extra changes from original commits, probably from auto format from IDE ?

**Review by margamanterola — APPROVED** [review]
Approved for my change

**Review by jschwinger233 — APPROVED** [review]
Looks good for my PR.

**Comment by doniacld** [comment]
I need to fix C code: https://github.com/cilium/cilium/actions/runs/6271820562/job/17032148237?pr=28251

File: Documentation/Makefile [hunk 2]
```diff
@@ -57,19 +59,26 @@ DOCKER_CTR := $(DOCKER_CTR_BASE) \
...
 update-cmdref: builder-image cilium-build ## Update the command reference documents (agent, bugtool, operators, etc.).
 	@$(ECHO_GEN)cmdref
 	-$(QUIET)rm -rf cmdref/cilium*.md
 	$(QUIET)$(DOCKER_RUN) ./update-cmdref.sh
 
+.PHONY: update-crdlist
+update-crdlist: ## Update the list of CRDs.
+	@$(ECHO_GEN)crdlist
+	make -C ../ generate-crd-docs
+
```
Comment (qmonnet) [inline Documentation/Makefile]: No, this is not relevant on v1.13

File: Documentation/Makefile [hunk 3]
```diff
@@ -83,6 +92,13 @@ ifeq ($(V),0)
 SPHINX_OPTS += -q
 endif
 
+.PHONY: api-flaggen
+api-flaggen: ## Update the table of API flags restrictions.
+	@$(ECHO_GEN) api-flags
+	$(QUIET)$(GO) run $(ROOT_DIR)/tools/api-flaggen \
+		2>/dev/null \
+		> configuration/api-restrictions-table.rst
+
```
Comment (qmonnet) [inline Documentation/Makefile]: This target is not relevant either on v1.13, just drop this chunk please

File: Documentation/Makefile [hunk 4]
```diff
@@ -61,43 +61,30 @@ DOCKER_RUN := $(DOCKER_CTR) $(DOCS_BUILDER_IMG)
 
 ##@ Auto-generated Contents Updates and Validation
 
+.PHONY: api-flaggen
+api-flaggen: ## Update the table of API flags restrictions.
+	@$(ECHO_GEN) api-flags
+	$(QUIET)$(GO) run $(ROOT_DIR)/tools/api-flaggen \
+		2>/dev/null \
+		> configuration/api-restrictions-table.rst
```
Comment (qmonnet) [inline Documentation/Makefile]: Same: `api-flaggen` not used in 1.13 docs

File: Documentation/Makefile [hunk 5]
```diff
@@ -120,9 +107,27 @@ $(HELM_VALUES): FORCE
...
+	@$(ECHO_CHECK) $(HELM_VALUES)
+	$(QUIET) ./check-helmvalues.sh
+	@$(ECHO_CHECK) examples
+	$(QUIET)$(DOCKER_RUN) ./check-examples.sh
+	@$(ECHO_CHECK) codeowners.rst
+	$(QUIET) ./check-codeowners.sh
+	@$(ECHO_CHECK) configuration/api-restrictions-table.rst
+	$(QUIET) ./check-flaggen.sh
+	@$(ECHO_CHECK) crdlist.rst
+	$(QUIET) ./check-crdlist.sh
```
Comment (qmonnet) [inline Documentation/Makefile]: These were not in the block we moved for 1.13, and should be skipped please

File: .github/workflows/conformance-gateway-api.yaml (hunk 1 above)
Comment (sayboras) [reply .github/workflows/conformance-gateway-api.yaml]: Can you help to revert some other changes? last time, I and Nicolas were struggling with all indentations.

File: .github/workflows/lint-bpf-checks.yaml [hunk 6]
```diff
@@ -42,6 +43,9 @@ jobs:
               - 'contrib/coccinelle/**'
             bpf-tests-runner:
               - 'test/bpf_tests/**'
+              - 'pkg/bpf/**'
```
Comment (qmonnet) [inline .github/workflows/lint-bpf-checks.yaml]: This line is not in the original commit. Please add a note in the commit description.

File: .github/workflows/lint-bpf-checks.yaml [hunk 7]
```diff
@@ -78,14 +82,14 @@ jobs:
     outputs:
       clang_dir: ${{ steps.set_dir.outputs.clang_dir }}
     steps:
-    - name: Set directory
-      id: set_dir
-      run: echo "clang_dir=$HOME/.clang" >> $GITHUB_OUTPUT
+      - name: Set directory
+        id: set_dir
+        run: echo "clang_dir=$HOME/.clang" >> $GITHUB_OUTPUT
```
Comment (qmonnet) [inline .github/workflows/lint-bpf-checks.yaml]: This chunk is also not part of the backported commit, looks good but please add a note.

File: Documentation/requirements.txt [hunk 8]
```diff
@@ -1,72 +1,70 @@
...
+rstcheck-core==1.1.1
+shellingham==1.5.3
 six==1.16.0
 snowballstemmer==2.2.0
-sphinx-mdinclude==0.5.3
-sphinxcontrib-applehelp==1.0.2
-sphinxcontrib-devhelp==1.0.2
-sphinxcontrib-htmlhelp==2.0.0
+sphinx_mdinclude==0.5.3
+sphinxcontrib-applehelp==1.0.7
```
Comment (qmonnet) [inline Documentation/requirements.txt]: I note you don't have the same version as on `main` for this one and a few others. I'm fine if you regenerated with `make -C Documentation update-requirements`, but I hit some issues with the latest versions of some add-ons for the Netlify preview on `main`. If this happens here as well, we'd probably better copy the file from `main` directly.

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thanks Donia! Please find some comments on the docs backports - Let me know if this makes too many conflicts to handle

File: .github/workflows/lint-bpf-checks.yaml (hunk 7 above)
Comment (doniacld) [reply .github/workflows/lint-bpf-checks.yaml]: Updated.

File: .github/workflows/lint-bpf-checks.yaml (hunk 6 above)
Comment (doniacld) [reply .github/workflows/lint-bpf-checks.yaml]: Oups! Removed.

File: Documentation/requirements.txt (hunk 8 above)
Comment (doniacld) [reply Documentation/requirements.txt]: The command `make -C Documentation update-requirements` only updated the following lines:
```
pydantic==2.4.0
pydantic_core==2.10.0
```

Do you think I should copy from the `main` directly?

File: .github/workflows/conformance-gateway-api.yaml (hunk 1 above)
Comment (doniacld) [reply .github/workflows/conformance-gateway-api.yaml]: I fixed it manually and removed the formatting when saving file for the moment in VSCode.

File: .github/workflows/conformance-ingress.yaml [hunk 9]
```diff
@@ -183,7 +182,7 @@ jobs:
           METALLB_IP_START=$(echo ${KIND_NET_CIDR} | sed "s@0.0/16@255.200@")
           METALLB_IP_END=$(echo ${KIND_NET_CIDR} | sed "s@0.0/16@255.250@")
           METALLB_IP_RANGE="${METALLB_IP_START}-${METALLB_IP_END}"
-          
+
```
Comment (doniacld) [inline .github/workflows/conformance-ingress.yaml]: @sayboras I could not fix those ones...

File: .github/workflows/conformance-gateway-api.yaml (hunk 1 above)
Comment (sayboras) [reply .github/workflows/conformance-gateway-api.yaml]: Thanks a lot :)

**Review by sayboras — APPROVED** [review]
Thanks and looks good for my commit :heavy_check_mark:

File: .github/workflows/conformance-ingress.yaml (hunk 9 above)
Comment (sayboras) [reply .github/workflows/conformance-ingress.yaml]: All good :heavy_check_mark:

**Review by viktor-kurchenko — APPROVED** [review]

**Review by lmb — APPROVED** [review]

**Comment by giorio94** [comment]
The coccicheck failures are being addressed in https://github.com/cilium/cilium/pull/28279

**Review by julianwiedmann — APPROVED** [review]

File: Documentation/requirements.txt (hunk 8 above)
Comment (qmonnet) [reply Documentation/requirements.txt]: Yes, let's copy it please, this should address the issue with Netlify.

I've been trying to keep the docs framework to the same version on all branches (this is why I asked for backports for these PRs), apart from the few content-specific Makefile targets we've mentioned earlier. So the dependencies should be the same as for `main`, and we're using the same image in CI anyway so it only makes sense to have identical `requirements.txt` files.

**Comment by qmonnet** [comment]
Thanks for fixing the issues on documentation side. It should be all good after we pick the `requirements.txt` from `main`. The backports for the other 2 doc PRs look good, I ticked the boxes above.

File: Documentation/requirements.txt (hunk 8 above)
Comment (giorio94) [reply Documentation/requirements.txt]: Thanks! I've updated the cherry-picked commit picking `Documentation/requirements.txt` verbatim from the main branch.

**Review by qmonnet — APPROVED** [review]
Looks good from my side, thanks!

This time the Netlify failure was apparently a transient issue (failure to download Node.js :upside_down_face:), I re-triggered [and it passed](https://app.netlify.com/sites/docs-cilium-io/deploys/6512e6706343a1105d2ac3da).

**Comment by giorio94** [comment]
> This time the Netlify failure was apparently a transient issue (failure to download Node.js 🙃), I re-triggered [and it passed](https://app.netlify.com/sites/docs-cilium-io/deploys/6512e6706343a1105d2ac3da).

Thanks!

**Comment by giorio94** [comment]
Additionally backported 4d699d8038e34e207c69dde73d33dc58d5060b1c, which was previously missing. /cc @qmonnet

**Comment by qmonnet** [comment]
> Additionally backported [4d699d8](https://github.com/cilium/cilium/commit/4d699d8038e34e207c69dde73d33dc58d5060b1c), which was previously missing. /cc @qmonnet

Not sure what happened, I'm pretty sure this commit was present when I did my first review for this PR. Anyway. Related backport looks good :+1:.

**Comment by giorio94** [comment]
Rebased onto the tip of v1.13 to pick the coccicheck fixes.

**Comment by giorio94** [comment]
/test-backport-1.13

**Review by joamaki — APPROVED** [review]

**Pull request merged**